UKHouses/__init__.py

```diff
@@ -32,74 +31,73 @@
-    #### 3. perform CRUD to later move to crud.py ###
-    #################################################
-
 
 """
 Resource fields for marshalling
 """
 page_fields = {
-    'Id': fields.String, 'code': fields.String, 'price': fields.String, 'sold date': fields.String,
+    'Id': fields.String, 'code': fields.String, 'price': fields.String, 'date': fields.String,
     'postcode': fields.String, 'Property Type': fields.String, 'New built?': fields.String,
     'Estate Type': fields.String, 'House/flat number': fields.String, 'Street Address': fields.String,
     'town': fields.String, 'district': fields.String, 'county': fields.String
 }
 
 
-# RESOURCES
+# RESOURCES - external repr of an Entity, it maps endpoints
+class Home(Resource):
+    def get(self):
+        return render_template('index.html'), 200
+
+
 class House(Resource):
     def get(self, Id):
-        tw = HousePricesModel.query.filter_by(Id=Id).first()
-        if tw:
-            return tw.json()
+        houseId = HousePricesModel.query.filter_by(Id=Id).first()
+        if houseId:
+            return {"house": houseId.json()}
         else:
             return {'Id': None}, 404
 
 
 class HouseList(Resource):
     def get(self):
-        houseprices = HousePricesModel.query.all()
-        return pagination.paginate([hp.json() for hp in houseprices], page_fields)
-
+        # print(request.args)
+        args = request.args
+        from_date = args.get('from')
+        to_date = args.get("to")
+        if request.method == 'GET' and 'from' in args is not None:
+            connection = sqlite3.connect(db_path)
+            cursor = connection.cursor()
+            query = "SELECT * FROM HousePrices WHERE date BETWEEN ? AND ? ORDER BY date ASC"
+            result = cursor.execute(query, (from_date, to_date,))
+            housesInDateRange = []
+            for row in result:
+                housesInDateRange.append(
+                    {"Id": row[0], "code": row[1], "price": row[2], "date": row[3], "postcode": row[4],
+                     "Property Type": row[5],
+                     "New built?": row[6], "Estate Type": row[7], "House/flat number": row[8], "Street Address": row[9],
+                     "town": row[10]
+                        , "district": row[11], "county": row[12]})
+            connection.close()
+            return pagination.paginate(housesInDateRange, page_fields)
 
-class Home(Resource):
-    def get(self):
-        # return {'hello':'world'}
-        return render_template('index.html'), 200
+        else:
+            houseprices = HousePricesModel.query.all()
+            return pagination.paginate([hp.json() for hp in houseprices], page_fields)
 
-        ####################
-        #### 3. end CRUD ###
-        ####################
 
+api.add_resource(Home, '/', '/index')
+api.add_resource(HouseList, '/houses')  # this is like: @app.route('/houses')
+api.add_resource(House, '/house/<string:Id>')
```

Comment:
The `/house` vs `/houses` debate is an ongoing conversation in the world of software. Nothing wrong with the approach taken here. My personal approach is to use plural `/houses` for both list and retrieve endpoints as, in my mind at least, they represent the same resource so should be the same endpoint. Although ... perhaps as it's easier for me to remember 😅.